pkg/apis/autoscaling/types.go

```diff
@@ -95,6 +95,68 @@ type HorizontalPodAutoscalerSpec struct {
 	// more information about how each type of metric must respond.
 	// +optional
 	Metrics []MetricSpec
+
+	// Behavior contains settings to specify the scaling behavior both while scaling up
+	// and scaling down. It also is used to customize other aspects of scaling like the
+	// scale down stabilization window
+	// +option
+	Behavior *HorizontalPodAutoscalerBehavior
+}
+
+// HorizontalPodAutoscalerBehavior configures a scaling behavior for Up and Down direction
+// (scaleUp and scaleDown fields respectively)
+type HorizontalPodAutoscalerBehavior struct {
+	// The scaling behavior while scaling up. Contains a list of policies and the policy selector.
+	// +optional
+	ScaleUp *HPAScalingRules
+	// The scaling behavior while scaling down. Contains a list of policies and the policy selector.
+	// +optional
+	ScaleDown *HPAScalingRules
+}
+
+// ScalingPolicySelect is used to specify which policy should be used while scaling in a certain direction
+type ScalingPolicySelect string
+
+const (
+	// MaxPolicySelect selects the policy with the highest possible change
+	MaxPolicySelect ScalingPolicySelect = "max"
+	// MinPolicySelect selects the policy with the lowest possible change
+	MinPolicySelect ScalingPolicySelect = "min"
+)
+
+// HPAScalingRules configures the scaling policy and the policy selector
+type HPAScalingRules struct {
+	// StabilizationWindowSeconds is the number of seconds for which past recommendations should be considered while
+	// scaling down. This prevents scale down of the autoscaling target in case the load fluctuates.
+	// +optional
+	StabilizationWindowSeconds *int32
+	// SelectPolicy is used to specify which policy should be selected after evaluation.
+	// +optional
+	SelectPolicy *ScalingPolicySelect
```

Comment:
I'm glad you made this a dedicated type instead of a `*string` as in the KEP.  
![image](https://user-images.githubusercontent.com/1103629/66808968-ace81080-ef2c-11e9-8be0-1124e7a603af.png)
You might consider updating the KEP to match.

Reply:
We discussed that and decided to merge the KEP as it was because KEP is just a proposal, not the documentation/API change itself.
If you feel like we should update the KEP, we can definitely do it.

Reply:
After everything is done, it would be nice.  But not necessary.  :)

Reply:
We'll update the KEP as soon as the API is stabilized! :)